### ISSUE TITLE
LinphoneService won't register on 64-bit phones (tried only arm64 though)

### ISSUE THREAD

**Issue by gajicm93** [BelledonneCommunications/linphone-android#38]
Hi, I'm doing some testing on a custom app I built by including your precompiled Android SDK (jar + SOs), and also including partial application source code into my app. The problem is that everything works perfectly for me on 32-bit phones, service, calls and everything, but when I run the app on 64-bit phones, the service just shows "failed to register". Do you have any idea what the problem might be? Thanks!

**Comment by DatBewar** [comment]
Hello,
I use Linphone on 64-bit phone and I do not have problem.
Can you check log to see if any problem occurs?

**Comment by gajicm93** [comment]
LinphoneProxyConfig [0xcd012000] about to register (LinphoneCore version: 3.10.2-580-g942f203)

get_jni_env(): thread successfuly attached

bellesip_wake_lock_acquire(): Android wake lock acquired [ref=0x20080e]

channel [0xcb24a420]: starting transaction background task with id=[20080e].

(ERROR LOG) No listening point matching for [udp://10.0.0.1:5060]

(ERROR LOG) belle_sip_client_transaction_send_request(): no channel available

Changing [client] [REGISTER] transaction [0xcb24a420], from state [INIT] to [TERMINATED]

Client REGISTER transaction [0xcb24a420] terminated

Unhandled transaction terminated [0xcb24a420]

channel [0xcb24a420]: ending transaction background task with id=[20080e].

bellesip_wake_lock_release(): Android wake lock released [ref=0x20080e]

Proxy config [0xcd012000] for identity [sip:300301@10.0.0.1] moving from state [LinphoneRegistrationNone] to [LinphoneRegistrationFailed] on core [0xca942e00]

New registration state [RegistrationFailed]

Linphone core [0xca942e00] notified [registration_state_changed]

**Comment by gajicm93** [comment]
Also just to be clear again, when I install stock Linphone app, it works on my 64bit phone, but my own app built with Linphone SDK doesn't.. While it works on 32-bit normally.. Could I be missing something from the gradle file or something?

**Comment by gajicm93** [comment]
Did some additional checking and logging, and yes, essentially this is the problem:

```
No listening point matching for [udp://10.0.0.1:5060]
belle_sip_client_transaction_send_request(): no channel available
```

On a 32-bit phone, I get:

```
belle_sip_client_transaction_send_request(): waiting channel to be ready
bellesip_wake_lock_acquire(): Android wake lock acquired [ref=0x100c6a]
channel [0xb8e3e450]: starting send background task with id=[100c6a].
channel [0xb8e3e450]: starting resolution of 10.0.0.1
channel 0xb8e3e450: state RES_IN_PROGRESS
```

... and so on.. and registration succeeds.

**Comment by gajicm93** [comment]
Ok, this got fixed with moving to new 64bit SDK.

But interesting thing, I Googled a bit, and even managed to fix it on the old one by setting useRandomPort inside LinphonePreferences.

**Comment by gajicm93** [comment]
Still fails even with 64bit SDK if useRandomPort not set..

Only useRandomPort fixes it for real.

**Comment by jalpesh101291** [comment]
Please Visit my blog http://linphonecode.blogspot.in please atleast one time visit i have ios and android source co

**Comment by lichao3140** [comment]
Hi,How do you solve this problem？
No listening point matching for [udp://120.25.126.228:5060 ]
belle_sip_client_transaction_send_request(): no channel available
THX

**Comment by gajicm93** [comment]
@lichao3140 did you try to set useRandomPort on LinphoneCore instance?

**Comment by lichao3140** [comment]
Yes, This bug appear Android 5.0, I have tested Android 4.4 and Android 5.1 were not this bug.

**Comment by lichao3140** [comment]
@gajicm93 Yes, This bug appear Android 5.0, I have tested Android 4.4 and Android 5.1 were not this bug.

**Comment by lnpusong123** [comment]
Well, this bug also exist on Nexus6 Android7.1.2. When i set useRandom port, it was solved.

**Comment by xcy396** [comment]
@lnpusong123 
Hello, I cannot find the method to set random port in LinphoneCore, i want to know how to set that, thank you!

**Comment by gajicm93** [comment]
@xcy396 Sorry it's here https://github.com/BelledonneCommunications/linphone-android/blob/3c53044a44619c3407612b4fb0f9af0e58c4dce1/src/android/org/linphone/LinphonePreferences.java#L1012

**Comment by xcy396** [comment]
@gajicm93 
thanks, in my current project I don't have the LinphonePreferences class, it woks well but cannot register on Nexus6 and Nexus6p . Now I add the useRandomPort method in the LinphoneManager class manually, and set use random port, the project runs perfect! Thanks very much!

```java
LinphoneUtils.getConfig(mServiceContext).setBool("app", "random_port", true);
setSipPort(-1);
```

**Comment by viv578ek** [comment]
1. **2019-07-11 16:49:18:820 ortp-error-No listening point matching for [udp://xxxx.xxxxx.xxxx:5060]
2019-07-11 16:49:18:820 ortp-error-belle_sip_client_transaction_send_request(): no channel available
2019-07-11 16:49:18:820 ortp-message-Changing [client] [REGISTER] transaction [0x600003386340], from state [INIT] to [TERMINATED]**

2. **"2019-07-11 16:49:18:665 ortp-message-Sal use rport [enabled]
2019-07-11 16:49:18:665 ortp-message-Root ca path disabled
2019-07-11 16:49:18:665 ortp-message-sal_unlisten_ports done
2019-07-11 16:49:18:665 ortp-message-Creating listening point [0x6000028b6880] on [sip:[::0]:5066;transport=TCP]"**


3. **2019-07-11 16:49:18:666 ortp-error-TCP bind() failed for ::0 port 5066: Address already in use
2019-07-11 16:49:18:666 ortp-message-Listening point [0x6000028b6880] on [sip:[::0]:5066;transport=TCP] destroyed
2019-07-11 16:49:18:666 ortp-warning-Could not start tcp transport on port 5066, maybe this port is already used.**

I am also facing related issues I want to use 5066 port but it has shown [udp://xxxx.xxxxx.xxxx:5060]. It has to use a random port 5060.
The server is also required 5066 port when user register. This port(5060) a user is not registered successfully. fail response

Please help me shorty.

**Comment by Viish** [comment]
@viv578ek Read this: https://github.com/BelledonneCommunications/linphone-android/issues/710. Also don't comment on old tickets that have no link whatsoever, instead open a new one.
